src/truncate.jl

```diff
@@ -49,33 +49,94 @@ insupport(d::Truncated{D,Union{Discrete,Continuous}}, x::Real) where {D<:Univari
 
 quantile(d::Truncated, p::Real) = quantile(d.untruncated, d.lcdf + p * d.tp)
 
-for f in [:pdf, :logpdf, :cdf, :logcdf, :ccdf, :logccdf]
-    @eval ($f)(d::Truncated, x::Int) = ($f)(d, float(x))
+function _pdf(d::Truncated, x::T) where {T<:Real}
+    if d.lower <= x <= d.upper
+        pdf(d.untruncated, x) / d.tp
+    else
+        zero(T)
+    end
+end
+
+function pdf(d::Truncated{D}, x::T) where {D<:ContinuousUnivariateDistribution, T<:Real}
+    _pdf(d, float(x))
+end
+
+function pdf(d::Truncated{D}, x::T) where {D<:DiscreteUnivariateDistribution, T<:Real}
+    isinteger(x) || zero(float(T))
+    _pdf(d, x)
+end
+
+function pdf(d::Truncated{D}, x::T) where {D<:DiscreteUnivariateDistribution, T<:Integer}
```

Comment:
I don't think this method is needed, since `isinteger(::Integer)` will be determined to be `true` at compile time in the method above.

Reply:
its main purpose it allowing dispatch

Reply:
How so?

Reply:
there is a method generic on discrete distribution types and Integer x, so to avoid ambiguous method definition, you need to implement `pdf(::D, x::Integer)`

Reply:
so you need this method regardless. Given that it's here, better not put the integer check here